### PR TITLE
VIH-9323 - Add custom domain for SignalR

### PR DESCRIPTION
**Pull request by thomaswatsonv1**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-9323

### Change description ###
Add managed identity, custom domain for signal R and AzAPI provider

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: pipeline-steps/terraform-plan.yaml [hunk 1]
```diff
@@ -89,6 +89,20 @@ steps:
           Write-Host "##vso[task.setvariable variable=SUBSCRIPTION_ID;isOutput=true]$subscriptionId"
           Write-Host "Subscription ID: $subscriptionId"
 
+  - ${{ if and( eq(parameters.activity_name, 'Video_Hearing_Core'), eq(parameters.environment, 'stg') ) }}:      
+    - task: AzureCLI@2
```
Comment (chris-pateman) [inline pipeline-steps/terraform-plan.yaml:93]: Can this be swapped to use the template please
https://github.com/hmcts/azure-devops-templates/blob/master/templates/Azure/Common/get-service-connection-details.yaml

File: terraform/09-video-hearing-core/20-main.tf [hunk 2]
```diff
@@ -258,15 +259,46 @@ module "storage" {
...
+  custom_domain_name  = var.signalr_custom_domain_name
+  key_vault_cert_name = var.environment == "stg" || var.environment == "prod" ? "wildcard-hearings-reform-hmcts-net" : "wildcard-${var.environment}-platform-hmcts-net"
+  key_vault_uri       = var.environment == "stg" ? data.azurerm_key_vault.acmekvstg[0].vault_uri : data.azurerm_key_vault.acmekv[0].vault_uri
+  tags                = local.common_tags
+}
 
-  tags = local.common_tags
+resource "azurerm_role_assignment" "acmmekv_access_policy" {
+  count                = var.environment != "stg" ? 1 : 0
+  role_definition_name = "Key Vault Administrator"
```
Comment (chris-pateman) [inline terraform/09-video-hearing-core/20-main.tf:291]: It should not be Admin access.
please us "Key Vault Secrets User"

File: terraform/09-video-hearing-core/modules/SignalR/init.tf [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Required for This Provider / Module.
+terraform {
+  required_providers {
```
Comment (chris-pateman) [inline terraform/09-video-hearing-core/modules/SignalR/init.tf:3]: do we need a resource for AZ commands or can we just use Null Resources?
Is this well supported? Secure? performance?

**Review by chris-pateman — CHANGES_REQUESTED** [review]

File: config/terraform/application/vars/09-video-hearing-core/ithc.tfvars [hunk 4]
```diff
@@ -3,3 +3,5 @@ location = "uksouth"
 vh_tenant_id     = "replace_vh_tenant_id"
 vh_client_id     = "replace_vh_client_id"
 vh_client_secret = "replace_vh_client_secret"
+
+signalr_custom_domain_name = "signalr.ithc.platform.hmcts.net"
```
[the remark this answers is not in the document]
Comment (v1ferrarij) [reply config/terraform/application/vars/09-video-hearing-core/ithc.tfvars:7]: wont work, there is 3 different URL patterns to use.

File: terraform/09-video-hearing-core/modules/SignalR/init.tf (hunk 3 above)
Comment (v1ferrarij) [reply terraform/09-video-hearing-core/modules/SignalR/init.tf:3]: required for this module.

File: pipeline-steps/terraform-plan.yaml (hunk 1 above)
Comment (v1ferrarij) [reply pipeline-steps/terraform-plan.yaml]: removed.

File: terraform/09-video-hearing-core/20-main.tf [hunk 5]
```diff
@@ -258,15 +259,46 @@ module "storage" {
...
 
+data "azurerm_key_vault" "acmekv" {
+  count               = var.environment != "stg" ? 1 : 0
+  name                = "acmedtssds${var.environment}"
+  resource_group_name = "sds-platform-${var.environment}-rg"
+}
+
+# Staging is using the hearings.reform.hmcts.net wildcard which is only available in the Production KV.
+# Hence requirement for this logic.
+data "azurerm_key_vault" "acmekvstg" {
```
[the remark this answers is not in the document]
Comment (v1ferrarij) [reply terraform/09-video-hearing-core/20-main.tf]: removed.

File: terraform/09-video-hearing-core/20-main.tf (hunk 2 above)
Comment (v1ferrarij) [reply terraform/09-video-hearing-core/20-main.tf]: resolved.

File: terraform/09-video-hearing-core/10-inputs-required.tf [hunk 6]
```diff
@@ -885,4 +885,29 @@ variable "cvp_client_secret" {
 variable "cvp_subscription_id" {
   description = "Client Subscription ID of the CVP SP"
   type        = string
-}
\ No newline at end of file
+}
+
+variable "signalr_custom_domain_name" {
```
[the remark this answers is not in the document]
Comment (v1ferrarij) [reply terraform/09-video-hearing-core/10-inputs-required.tf:890]: wont work, there is 3 different URL patterns to use.

File: terraform/10-video-hearing-wowza/modules/wowza/lb-public.tf [hunk 7]
```diff
@@ -25,21 +26,20 @@ resource "azurerm_lb" "wowza-public" {
   tags = var.tags
 }
 
-resource "azurerm_lb_probe" "wowza_rtmps-public" {
+resource "azurerm_lb_backend_address_pool" "wowza-public" {
   loadbalancer_id = azurerm_lb.wowza-public.id
```
Comment (chris-pateman) [inline terraform/10-video-hearing-wowza/modules/wowza/lb-public.tf:30]: please do wowza changes in another PR to keep things seperate

**Review by chris-pateman — CHANGES_REQUESTED** [review]
